### ISSUE TITLE
New TTree(char *, char*) segfaults with PhysicsTools.PythonAnalysis.LumiList

### ISSUE THREAD

**Issue by odysei** [cms-sw/cmssw#18906]
A step-by-step procedure (inspired by https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookWriteFrameworkModule) to make a toy example that segfaults:

```
scram project CMSSW_9_2_X_2017-05-23-1100
cd CMSSW_9_2_X_2017-05-23-1100/src
cmsenv
mkdir Demo
cd Demo
mkedanlzr DemoAnalyzer
cd DemoAnalyzer
echo "import PhysicsTools.PythonAnalysis.LumiList as LumiList" > python/test.py
cat python/ConfFile_cfg.py | sed "s:myfile.root:/afs/cern.ch/cms/Tutorials/TWIKI_DATA/TTJets_8TeV_53X.root:" >> python/test.py
cat plugins/DemoAnalyzer.cc | \
    sed "s:\#include <memory>:\#include <memory>\n\#include \"TTree.h\":" | \
    sed "s:   usesResource(\"TFileService\");:   usesResource(\"TFileService\");\n    TTree *t = new TTree(\"test\", \"test\");\n    delete t;:" > temp.cc
mv temp.cc plugins/DemoAnalyzer.cc
scram b
cmsRun python/test.py
```

**Comment by Dr15Jones** [comment]
assign analysis

**Comment by davidlt** [comment]
@Dr15Jones this should be assigned to Core, see https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PythonAnalysis/python/LumiList.py It's basically FWCore.PythonUtilities.LumiList.

**Comment by davidlt** [comment]
I looked a bit into this today using  the original source code. TTree segfaults because gStyle is 0x0. gStyle becomes 0x0 because ROOT cleans up it's global environment (i.e. EndOfProcessCleanups is triggered). A quick look suggest that this only should happen once you are done with ROOT.

**Comment by Dr15Jones** [comment]
assign core

**Comment by Dr15Jones** [comment]
cmsRun does shutdown the python interpreter before starting. However, I would not expect ROOT to have cared about that.

**Comment by davidlt** [comment]
I replaced `import PhysicsTools.PythonAnalysis.LumiList as LumiList` with `import FWCore.PythonUtilities.LumiList as LumiList` and it worked just fine (no segfault).

**Comment by Dr15Jones** [comment]
Maybe just a shift in memory addresses?

**Comment by davidlt** [comment]
Just toying around, but just having `PhysicsTools/PythonAnalysis/python/LumiList.py` as empty file still causes segfault.

**Comment by tvami** [comment]
+analysis
 * see https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PythonAnalysis/python/LumiList.py#L6

**Comment by smuzaffar** [comment]
usage of `PhysicsTools.PythonAnalysis.LumiList` has been deprecated . I am closing this issue, feel free to open a new issue if something still needs to be addressed
